examples/pie_and_polar_charts/pie_and_donut_labels.py

```diff
@@ -0,0 +1,121 @@
+# subplots. To have some space for the big cake, we use
+# :meth:`subplots_adjust <matplotlib.figure.subplots_adjust>` to adjust the
+# spacings.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+fig, (ax, ax2) = plt.subplots(nrows=2, figsize=(6, 6))
+fig.subplots_adjust(top=0.92, bottom=0.08)
+plt.setp((ax, ax2), aspect="equal")
+
+
+###############################################################################
+# Now it's time for the pie. Starting with a pie recipe, we create the data
+# and a list of labels from it.
+#
+# We can `provide a function to the ``autopct`` argument, which will expand
+# automatic percentage labeling by the absolute values, which we calculate 
+# back from realtive data and the known sum of all values.
+#
+# We then create the pie and store the tuple of returned objects in a
+# variable ``pie``.
+# The first returned element of that tuple is a list of the wedges. Those are
+# :class:`matplotlib.patches.Wedge <matplotlib.patches.Wedge>` patches, which
+# can directly be used as the handles for a legend. We can use the 
+# legend's ``bbox_to_anchor`` argument to position the legend outside of
+# the pie. Here we use the axes coordinates ``(1, 0, 0.5, 1)`` together 
+# with the location ``"center left"``; i.e.
+# the left central point of the legend will be at the left central point of the
+# bounding box, spanning from ``(1,0)`` to ``(1.5,1)`` in axes coordinates. 
+
+
+recipe = ["375 g flour",
+          "75 g sugar",
+          "250 g butter",
+          "300 g berries"]
+
+data = [float(x.split()[0]) for x in recipe]
+ingredients = [x.split()[-1] for x in recipe]
+
+def func(pct, allvals):
+    absolute = int(pct/100.*np.sum(allvals))
+    return "{:.1f}%\n({:d} g)".format(pct, absolute)
+
+pie = ax.pie(data, autopct=lambda pct: func(pct, data),
+             textprops=dict(color="w"))
+
+ax.legend(pie[0], ingredients,
+          title="Ingredients",
+          loc="center left",
+          bbox_to_anchor=(1, 0, 0.5, 1))
+
+plt.setp(pie[2], size=8, weight="bold")
+
+ax.set_title("Matplotlib bakery: A pie")
+
+
+
+###############################################################################
+# Now it's time for the donut. Starting with a donut recipe, we transcribe
+# the data to numbers (converting 1 egg to 50 g), and directly plot the pie.
+# The pie? Wait... it's going to be donut, is it not? 
+# Well, as we see here, the donut is a pie, having a certain ``width`` set to
+# the wedges, which is different from its radius. It's as easy as it gets.
+# This is done via the ``wedgeprops`` argument.
+#
+# Labeling the wedges is done via ``annotate``. We first create some
+# dictionaries of common properties, which we can later pass as keyword
+# argument. We then iterate over all wedges and for each
+#
+# * calcualte the angle of the wedge's center
+# * from that obtain the coordinates of the point at that angle on the 
+#   circonference
```

Comment:
circumference

Reply:
Sir! Conferences always turn in circles! :laughing: